docx/oxml/text/run.py

```diff
@@ -79,24 +79,22 @@ def add_drawing(self, inline_or_anchor):
 
     def clear_content(self):
         """
-        Remove all child elements except the ``<w:rPr>`` element if present.
+        Remove all child elements except the ``<w:rPr>`` and ``<w:footnoteReference>`` element if present.
```

Comment:
Why are these preserved?

Reply:
Okay, another connected question, why do we need to preserve footnote reference ids when clearing content?

Reply:
Hm, maybe we should change the function name, because clear content is actually for removing text content. The problem is in function `replace_chars` in Paragraphs, because this function when replacing run characters calls the clear content function that should only remove the textual information.